server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -872,6 +873,17 @@ public void writeVIntArray(int[] values) throws IOException {
         }
     }
 
+    public void writeBigIntArray(IntArray values) throws IOException {
+        // TODO do we need to specialize this for ReleasableIntArray
+        if (values.size() > Integer.MAX_VALUE / Integer.BYTES) {
+            throw new UnsupportedOperationException(); // NOCOMMIT What do?
+        }
+        writeVInt((int) values.size() * Integer.BYTES);
```

Comment:
We need to think about size here - the arrays are often oversized but we will have a precise size when we're ready to write. If we write the oversized array it could be twice as big.